mmaction/datasets/pipelines/loading.py

```diff
@@ -178,6 +178,61 @@ def __call__(self, results):
         return results
 
 
+@PIPELINES.register_module()
+class UntrimSampleFrames(object):
+    """Sample frames from the untrimmed video.
+
+    Required keys are "filename", added or modified keys are "total_frames",
+    "frame_inds", "frame_interval" and "num_clips".
+
+    Args:
+        clip_len (int): The length of sampled clips. Default: 1. (5 for Flow)
+        frame_interval (int): Temporal interval of adjacent sampled frames.
+            Default: 16.
+        start_index (int): Specify a start index for frames in consideration of
+            different filename format. However, when taking videos as input,
+            it should be set to 0, since frames loaded from videos count
+            from 0. Default: 1.
+    """
+
+    def __init__(self, clip_len=1, frame_interval=16, start_index=1):
+
+        self.clip_len = clip_len
+        self.frame_interval = frame_interval
+        self.start_index = start_index
+
+    def __call__(self, results):
+        """Perform the SampleFrames loading.
+
+        Args:
+            results (dict): The resulting dict to be modified and passed
+                to the next transform in pipeline.
+        """
+        if 'total_frames' not in results:
+            # TODO: find a better way to get the total frames number for video
+            video_reader = mmcv.VideoReader(results['filename'])
```

Comment:
can you guarantee that the input is video?

Reply:
Yes, it's clear that only when you use rawframe_dataset, the key 'total_frames' will appear in the video_info. Otherwise, the input is a video.

Reply:
It seems not right, the class `XXX_Init` will add key `total_frames` in it. you can rebase master to get the latest codes.

Reply:
I see. Now that the 'total_frames' key becomes essential, removed the check.